common/djangoapps/student/views.py

```diff
@@ -708,7 +709,7 @@ def dashboard(request):
 
         for course_run in course_runs_for_course:
             enrollment_end = course_run.get('enrollment_end')
-            if not enrollment_end or enrollment_end > datetime.datetime.now(UTC):
+            if not (enrollment_end and dateutil.parser.parse(enrollment_end) <= datetime.datetime.now(UTC)):
                 enrollable_course_runs.append(course_run)
```

Comment:
This is slightly different from the above. The above allows course runs without a set enrollment end to go through.

Reply:
should be 
`if not enrollment_end or dateutil.parser.parse(enrollment_end) >= datetime.datetime.now(UTC)):`

Reply:
The original logic will evaluate to `True` if enrollment_end is falsy, OR if enrollment_end is in the future. The same is true of the new logic. The way I've rewritten may not be as clear as it was before though, so I'll change it.